### PR TITLE
Add autodoc of services and update services.auth for OAuth

### LINKED ISSUE #1064: update docs for oauth
- [x] update descriptions of how auth works
- [x] update service docs, pointing to new OAuth classes (existing docs / cookie-based auth still works)
- [ ] add new services.auth.OAuth classes to API docs

### PR DESCRIPTION
**Pull request by willingc**
Closes #1064

### CODE REVIEW

File: docs/rest-api.yml [hunk 1]
```diff
@@ -425,8 +425,9 @@ paths:
         '200':
           description: The user or service identified by the API token
           schema:
-            $ref: '#/definitions/User'
-            $ref: '#/definitions/Service'
+            $ref: 
+              - '#/definitions/User'
+              - $'#/definitions/Service'
```
Comment (minrk) [inline docs/rest-api.yml]: extra dollar sign?

Comment (willingc) [inline docs/rest-api.yml]: Yep. Good catch. I was having odd errors with RTD (npm build errors) all day so I may let this sit until morning.

**Comment by willingc** [comment]
[Rendered docs](http://test-jupyterhub.readthedocs.io/en/doc-services/index.html)

Removed the schema in `rest-api.yml` that was causing node errors in doc build.

**Comment by minrk** [comment]
Nice!

**Pull request merged**